src/app/services/app-state.service.ts

```diff
@@ -98,22 +103,33 @@ export class AppStateService {
 
       if (queryParams?.length) {
         const baseUrl = window.location.href.substr(0, window.location.href.indexOf("?"));
-
         window.history.replaceState(window.history.state, document.title, `${baseUrl}?${queryParams.join("&")}`);
       }
     }
-    else if (window.parent.location.search?.includes("rowId")) {
-      const messageJSON = {
-        updateCinchyURLParams:
-        {
-          rowId: cinchyId
-        }
-      };
-
-      window.parent.postMessage(JSON.stringify(messageJSON), '*');
-    }
+  }
 
+  updateConnectionQueryParams(id: number) {
+    const messageJSON = {
+      updateCinchyURLParams:
+      {
+        rowId: id
+      }
+    };
+    const message = JSON.stringify(messageJSON);
+    window.parent.postMessage(message, '*');
     sessionStorage.setItem("rowId", this._rowId ? this._rowId.toString() : "");
```

Comment:
There is no reason to modify the session storage in this function. Remove both of these statements.

Reply:
done